src/CMakeLists.txt

```diff
@@ -101,19 +101,27 @@ if (MSVC)
         string(REPLACE "/MD" "/MT" CMAKE_CXX_FLAGS_${build_type} "${CMAKE_CXX_FLAGS_${build_type}}")
         string(APPEND CMAKE_CXX_FLAGS_${build_type} " /GL")
 
+        # /GL requires /LTCG
+        string(APPEND CMAKE_EXE_LINKER_FLAGS_${build_type} " /LTCG")
+
         # TODO: replacing /INCREMENTAL leaves ":NO" on the command line, which screws up the link
         #       Figure out the best way to make these changes to the build and linker flags
         # string(REPLACE "/INCREMENTAL" "" CMAKE_EXE_LINKER_FLAGS_${build_type} "${CMAKE_EXE_LINKER_FLAGS_${build_type}}")
         # string(APPEND CMAKE_EXE_LINKER_FLAGS_${build_type} " /LTCG:INCREMENTAL /OPT:REF")
     endforeach()
 
+    # Always generate symbols for release builds
+    string(APPEND CMAKE_CXX_FLAGS_RELEASE " /Zi")
+    string(APPEND CMAKE_SHARED_LINKER_FLAGS_RELEASE " /DEBUG /OPT:REF /OPT:ICF")
```

Comment:
can we add /MAP too?

Reply:
Do we want that by default? I imagine it's not enormously useful in the general case and I assume increases build times

Reply:
I've been using it alot lately. Doesn't seem to increase build time by a significant amount that I've noticed.

Reply:
So I tried adding `/MAP` to the linker flags and I don't see a `.map` file getting produced. Is there anything additional that I need to do?

Reply:
nvm. Tried it again and it worked this time...